src/repositories/updater.rs

```diff
@@ -0,0 +1,366 @@
+use crate::error::Result;
+use crate::repositories::{GitHub, GitLab};
+use crate::utils::MetadataPackage;
+use crate::{db::Pool, Config};
+use chrono::{DateTime, Utc};
+use log::{debug, info, trace, warn};
+use once_cell::sync::Lazy;
+use postgres::Client;
+use regex::Regex;
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+pub trait RepositoryForge {
+    /// Result used both as the `host` column in the DB and to match repository URLs during
+    /// backfill.
+    fn host(&self) -> &str;
+
+    /// FontAwesome icon used in the front-end.
+    fn icon(&self) -> &'static str;
+
+    /// How many items we can query in one graphql request.
+    fn chunk_size(&self) -> usize;
```

Comment:
Ditto, maybe this should just be a constant?

Reply:
This one I'm not too sure. For example, you can have multiple different instances that match the gitlab API but they may have different chunk size. So just in case, I think it'd be better to keep it as a function just in case for the moment.

Reply:
Hmm, ok - it wouldn't work anyway because of the object safety requirement.